### PR TITLE
Add platform specific reset target

### PR DESCRIPTION
**Pull request by CiprianRegus**
## Pull Request Description

Fix 86d407b ("tools: scripts: Delete .vscode on reset").

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [ ] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [ ] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [ ] I have performed a self-review of the changes
- [ ] I have commented my code, at least hard-to-understand parts
- [ ] I have build all projects affected by the changes in this PR
- [ ] I have tested in hardware affected projects, at the relevant boards
- [ ] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

File: tools/scripts/generic.mk [hunk 1]
```diff
@@ -400,7 +400,7 @@ clean:
 
 # Remove the whole build directory
 PHONY += reset
-reset: clean_libs sdkclean
+reset: clean_libs sdkclean $(PLATFORM)_reset
```
Comment (buha) [inline tools/scripts/generic.mk]: remove the sdkclean dependency, in a separate commit

File: tools/scripts/aducm.mk [hunk 2]
```diff
@@ -273,6 +273,8 @@ copy_pinmux:
 
 update: copy_pinmux
 
+$(PLATFORM)_reset:
```
Comment (buha) [inline tools/scripts/aducm.mk:276]: make sure these empty ones don't generate make warnings for being empty, i recall seeing some empty rule warnings at some point but i'm not sure

**Comment by CiprianRegus** [comment]
Removed the reset dependency to sdkclean in a separate commit.

**Review by buha — APPROVED** [review]

**Comment by CiprianRegus** [comment]
Can someone confirm this fixes the issues with the Xilinx and mbed reset targets?

**Review by amiclaus — APPROVED** [review]

**Pull request merged**